test/e2e/create_test.go

```diff
@@ -67,11 +70,25 @@ func testWindowsNodeCreation(t *testing.T) {
 	}
 	testCtx, err := NewTestContext(t)
 	require.NoError(t, err)
+	// Create a private key secret with a randomly generated value. This is changed to the provided private key later in
+	// the test.
+	require.NoError(t, testCtx.createPrivateKeySecret(false), "could not create private key secret")
 
 	for _, test := range testCases {
 		if err := testCtx.createWindowsMachineSet(test.replicas, test.isWindows); err != nil {
 			t.Fatalf("failed to create Windows MachineSet %v ", err)
 		}
+		if test.isWindows {
+			// We need to cover the case where a user changes the private key secret before the WMCO has a chance to
+			// configure the Machine. In order to simulate that case we need to wait for the MachineSet to be fully
+			// provisioned and then change the key. Depending on timing and configuration flakes this will either cause
+			// the all Machines, or all Machines after the first configured Machines to hit this scenario.
```

Comment:
> Are you hoping that this happens after the Machine is Provisioned and before it is fully prepped? Won't this potentially result in flakes?

Yes, I am hoping to catch the second Machine which is why I wait for the Machine to enter provision state. Most likely the first one will be provisioned normally, as the key most likely not be changed before the first machine is queued. This can result in non-consistent test runs, such as if the first Machine has a reconcile error and needs to be re-queued, causing it to be deleted and recreated. But there shouldnt be a large negative impact on the test suite.

> Moreover where are you testing if the Machine got deleted in the scenario that the key is changed?

This PR doesn't deal with ensuring all the Machines were configured with the same key, it just solves the case where a Machine is stuck in Provisioned state due to a mismatched key. The fact that the Machine is configured is proof that the mismatched Machine was deleted. 

The next PR I have will deal with deleting configured Machines after the provided private key is changed.